dipy/testing/decorators.py

```diff
@@ -103,3 +109,109 @@ def _set_random_number_generator_wrapper(pytestconfig, *args, **kwargs):
         return _set_random_number_generator_wrapper
 
     return _set_random_number_generator
+
+
+def warning_for_keywords(from_version="1.10.0", until_version="2.0.0"):
+    """
+    Decorator to warn about keyword arguments passed as positional arguments
+    and handle version-based deprecation of functions.
+
+    Parameters
+    ----------
+    from_version : str
+        The version from which the warning should start.
+    until_version : str
+        The version until which the warning is applicable.
+
+    Returns
+    -------
+    function
+        The wrapped function that will issue a warning based
+        on the version.
+    """
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            current_version = dipy.__version__
+
+            def convert_positional_to_keyword(func, args, kwargs):
```

Comment:
The `func` parameter in `convert_positional_to_keyword` is not redundant because this function needs to know which function it should call with the modified arguments. While the outer decorator function knows about `func` because it is its parameter, `convert_positional_to_keyword` also needs to know about it, hence it is passed as an argument when calling `convert_positional_to_keyword`.

It also allows the helper function to be generic and reusable for any function that the decorator is applied to.